src/cache_utils.jl

```diff
@@ -1,6 +1,24 @@
 is_constant_cache(cache::OrdinaryDiffEqConstantCache) = true
 is_constant_cache(cache::OrdinaryDiffEqCache) = false
 is_constant_cache(cache::CompositeCache) = is_constant_cache(cache.caches[1])
+function is_constant_cache(cache::DefaultCache)
+    current = cache.current
+    if current == 1
+        is_constant_cache(cache.cache1)
+    elseif current == 2
+        is_constant_cache(cache.cache2)
+    elseif current == 3
+        is_constant_cache(cache.cache3)
+    elseif current == 4
+        is_constant_cache(cache.cache4)
```

Comment:
They should all always give the same answer? Being able to constant prop may be required for type stability?

Reply:
they won't give the same answer because if you look up one that hasn't been initialized, it can segfault. It also shouldn't be needed for type stability since all branches are type stable and return a `Bool`

Reply:
but it needs to constant prop to true/false to constant prop for the type choices after the function.

Reply:
Good catch. The new verison of this will now work.